tls/s2n_tls13_certificate_verify.c

```diff
@@ -125,15 +108,74 @@ int s2n_server_generate_unsigned_cert_verify_content(struct s2n_connection *conn
     GUARD(s2n_hash_free(&hash_copy));
 
     /* Concatenate the content to be signed/verified */
-    GUARD(s2n_stuffer_alloc(unsigned_content, hash_digest_length + s2n_server_cert_verify_header_length()));
+    GUARD(s2n_stuffer_alloc(unsigned_content, hash_digest_length + s2n_tls13_cert_verify_header_length(mode)));
     GUARD(s2n_stuffer_write_bytes(unsigned_content, S2N_CERT_VERIFY_PREFIX, sizeof(S2N_CERT_VERIFY_PREFIX)));
-    GUARD(s2n_stuffer_write_bytes(unsigned_content, S2N_SERVER_CERT_VERIFY_CONTEXT, sizeof(S2N_SERVER_CERT_VERIFY_CONTEXT)));
+
+    if(mode == S2N_CLIENT) {
+        GUARD(s2n_stuffer_write_bytes(unsigned_content, S2N_CLIENT_CERT_VERIFY_CONTEXT, sizeof(S2N_CLIENT_CERT_VERIFY_CONTEXT)));
+    } else {
+        GUARD(s2n_stuffer_write_bytes(unsigned_content, S2N_SERVER_CERT_VERIFY_CONTEXT, sizeof(S2N_SERVER_CERT_VERIFY_CONTEXT)));
+    }
+
     GUARD(s2n_stuffer_write_bytes(unsigned_content, digest_out, hash_digest_length));
 
     return 0;
 }
 
-uint8_t s2n_server_cert_verify_header_length()
+uint8_t s2n_tls13_cert_verify_header_length(s2n_mode mode)
 {
+    if (mode == S2N_CLIENT) {
+        return sizeof(S2N_CERT_VERIFY_PREFIX) + sizeof(S2N_CLIENT_CERT_VERIFY_CONTEXT);
+    }
     return sizeof(S2N_CERT_VERIFY_PREFIX) + sizeof(S2N_SERVER_CERT_VERIFY_CONTEXT);
 }
+
+int s2n_tls13_cert_verify_recv(struct s2n_connection *conn)
+{
+    /* Read the algorithm and update conn->secure.conn_sig_scheme */
+    GUARD(s2n_get_and_validate_negotiated_signature_scheme(conn, &conn->handshake.io, &conn->secure.conn_sig_scheme));
```

Comment:
is `conn->secure.conn_sig_scheme` used by both client/server?
should conn->handshake_params.server_sig_hash_algs be used?

Reply:
i believe it’s correct as it is set here when the server_hello is received and this message comes later https://github.com/awslabs/s2n/blob/master/tls/s2n_server_hello.c#L184